From 18bf77c220021b1086dc220719f2d3cc174652b3 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Tue, 12 Apr 2016 06:39:52 +1000
Subject: [PATCH 0538/1722] drm/amd/dal: drop get_scratch_lcd_state wrapper

This wrapper doesn't serve any point, lots of others functions
call directly in the bios_parser.c

Signed-off-by: Dave Airlie <airlied@redhat.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c           |  2 +-
 drivers/gpu/drm/amd/dal/dc/bios/bios_parser_helper.c    | 17 -----------------
 drivers/gpu/drm/amd/dal/include/bios_parser_interface.h |  3 ---
 3 files changed, 1 insertion(+), 21 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
index cca9f58..b2ccc66 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
@@ -154,7 +154,7 @@ void dc_bios_power_up(struct dc_bios *dcb)
 	struct bios_parser *bp = BP_FROM_DCB(dcb);
 
 	if (bp->lcd_scale == LCD_SCALE_UNKNOWN)
-		bp->lcd_scale = dal_bios_parser_get_scratch_lcd_scale(bp);
+		bp->lcd_scale = bp->bios_helper->get_scratch_lcd_scale(bp->ctx);
 #endif
 }
 
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_helper.c b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_helper.c
index e96314d..2037f15 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_helper.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_helper.c
@@ -97,20 +97,3 @@ bool dal_bios_parser_is_lid_status_changed(
 	return bp->bios_helper->is_lid_status_changed(
 			bp->ctx);
 }
-
-/**
-* dal_bios_parser_get_scratch_lcd_scale
-*
-* @brief
-*  get LCD Scale Mode from VBIOS scratch register
-*
-* @param
-*  NONE
-*/
-enum lcd_scale  dal_bios_parser_get_scratch_lcd_scale(
-	struct bios_parser *bp)
-{
-	return bp->bios_helper->get_scratch_lcd_scale(
-			bp->ctx);
-}
-
diff --git a/drivers/gpu/drm/amd/dal/include/bios_parser_interface.h b/drivers/gpu/drm/amd/dal/include/bios_parser_interface.h
index a177e15..e3b5ff5 100644
--- a/drivers/gpu/drm/amd/dal/include/bios_parser_interface.h
+++ b/drivers/gpu/drm/amd/dal/include/bios_parser_interface.h
@@ -49,7 +49,4 @@ bool dal_bios_parser_is_lid_open(
 	struct bios_parser *bp);
 bool dal_bios_parser_is_lid_status_changed(
 	struct bios_parser *bp);
-enum lcd_scale  dal_bios_parser_get_scratch_lcd_scale(
-	struct bios_parser *bp);
-
 #endif /* __DAL_BIOS_PARSER_INTERFACE_H__ */
-- 
2.7.4

